tests/containers/container_engine.pm

```diff
@@ -32,6 +32,88 @@ use containers::common;
 use containers::utils;
 use containers::container_images;
 
+sub basic_container_tests {
+    my %args = @_;
+    my $runtime = $args{runtime};
+    die "Undefined container runtime" unless $runtime;
+    my $image = "registry.opensuse.org/opensuse/tumbleweed";
+
+    ## Test search feature
+    validate_script_output("$runtime search --no-trunc --format 'table {{.Name}} {{.Description}}' tumbleweed", sub { m/Official openSUSE Tumbleweed images/ }, timeout => 300);
+
+    # Test pulling and display of images
+    script_retry("$runtime image pull $image", timeout => 600, retry => 3, delay => 120);
```

Comment:
```suggestion
    script_retry("$runtime image pull $image", timeout => 600, retry => 3, delay => 120, die => 1);
```
Pull failure check could be useful here, to distinguish from validation text checking

Reply:
I didn't note default. OK